ext/bcmath/bcmath.c

```diff
@@ -716,7 +721,8 @@ PHP_FUNCTION(bcround)
 		case PHP_ROUND_AWAY_FROM_ZERO:
 			break;
 		default:
-			zend_argument_value_error(3, "must be a valid rounding mode (PHP_ROUND_*)");
+			/* This is currently unreachable, but might become reachable when new modes are added. */
+			zend_argument_value_error(3, " is a rounding mode that is not supported");
```

Comment:
I don't see how this can happen now? As the value should always be a correct mode from the enum object, turning this into an assertion seems more logical.

Reply:
Cannot happen now (as the comment indicates), but can happen if additional cases are added to the `RoundingMode` enum in the future that cannot (easily) be supported with BCMath or where the implementation forgets to adjust BCMath.

This error message is thus intended to fail safely rather than crashing or returning bogus values.

Reply:
I would rather have this be an assertion and have a test in BC Math that cycles through all the cases of the enum to be sure we don't miss any new ones introduced.

Reply:
I've added an additional test based on `round_RoundingMode.phpt` that goes through all the enum cases, but I've preserved the error message instead of replacing it by an assert. The RoundingMode enum is owned by ext/standard, not ext/bcmath, so to me defensive programming makes sense there.